### ISSUE TITLE
There is error when init IE11 or execute javascript string with 'execute' command

### ISSUE THREAD

**Issue by tyler-tanliang** [webdriverio/webdriverio#2829]
## The problem

When run the chrome and firefox with same test cases, there is not any errors, but when I run it with IE11, there is error about execute command.
First, When init the driver, there is error as below:
```
[15:00:18]  COMMAND     POST     "/wd/hub/session/5dcc2111-7839-4ce8-8549-ffff8f5c93f0/**execute**"
[15:00:18]  DATA                {"script":"return (function getUserAgent() {\n  return window.navigator.userAgent;\n}).apply(null, arguments)","args":[]}
Error: Remote end send an unknown status code.
    at execute(<Function>) - D:\YGLCode\WebdriverTry\node_modules\wdio-sync\build\index.js:263:37
    at init() - D:\YGLCode\WebdriverTry\node_modules\webdriverio\build\lib\runner.js:235:55
```
When run the test step with browser.execute(javascriptString), there is error about execute command.

```
[15:02:40]  COMMAND     POST     "/wd/hub/session/2adacae9-dfd7-4270-aa18-fa82e2afdae9/**execute**"
[15:02:40]  DATA                {"script":"var url=document.URL;\nreturn url.indexOf('sample.com')>0;","args":[]}
saveScreenshot.the reason is:
{ Error: Remote end send an unknown status code.
    return url.indexOf('sample.com')>0;") -     at D:\YGLCode\WebdriverTry\node_modules\wdio-sync\build\at execute("var url=document.URL;
  details: undefined,
  message: '',
  type: 'RuntimeError',
  seleniumStack:
   { type: 'Unknown',
     message: 'Remote end send an unknown status code.',
     orgStatusMessage: '' } }
```
## Environment

* WebdriverIO version:4.12.0
* Node.js version:9.9.0
* [Standalone mode or wdio testrunner](http://webdriver.io/guide/getstarted/modes.html): wdio testrunner with selenium Grid.
* if wdio testrunner, running synchronous or asynchronous tests:Sync
* Additional wdio packages used (if applicable):allure,wdio-screenshot,visual-regression

## Details
When the visual-regression run at last, there is same error for the execute command.

```
 [15:02:40]  COMMAND     POST     "/wd/hub/session/2adacae9-dfd7-4270-aa18-fa82e2afdae9/execute"
[15:02:40]  DATA                {"script":"return (function scrollbars(enabled) {\n  if (enabled) {\n    document.documentElement.style.overflow = '';\n  } else {\n    document.documentElement.style.overflow = 'hidden';\n  }\n}).apply(null, arguments)","args":[false]}
{ Error: Remote end send an unknown status code.
    at saveDocumentScreenshot("D:\YGLCode\WebdriverTry\screenshots\temp\temp.png", image/png) -     at D:\YGLCode\WebdriverTry\node_modules\wdio-sync\build\at execute(<Function>, false) - D:\YGLCode\WebdriverTry\node_modules\wdio-sync\build\index.js:263:37
  details: undefined,
  message: '',
  type: 'RuntimeError',
  seleniumStack:
   { type: 'Unknown',
     message: 'Remote end send an unknown status code.',
     orgStatusMessage: '' } }
```

**Comment by christian-bromann** [comment]
This is most likely an issue with the IEDriver. There is nothing we can do to fix this.

**Comment by tyler-tanliang** [comment]
@christian-bromann , When I try to run test cases with C# and selenium Grid, the javascriptExecute can work well.
I think IEDriver can work for javascript execute.
Could you have a look again, Thanks very much!

**Comment by christian-bromann** [comment]
> I think IEDriver can work for javascript execute.

It does. Whatever wdio-screenshot is doing it causes an error. You might want to raise an issue in that project.

**Comment by tyler-tanliang** [comment]
Thanks, I will remove it and try again. If the bug is wdio-screenshot, I will submit there.
Thanks very much again!

**Comment by tyler-tanliang** [comment]
@christian-bromann ,When I only run WDIO with IEDriver and selenium Grid, there is error when run execute command.
the js is below:

```
describe('Test Describe', function(){
    it('Test it', function(){
        browser.url('http://webdriver.io');
        var result;
        try{
            console.log('start run exectue');
            result=browser.execute(function(a, b, c, d) {
                return a + b + c + d;
            }, 1, 2, 3, 4);
            console.log('finished execute');
        }
        catch(e)
        {
            console.log(e);
            console.log('resultis:'+result);
        }
    })
})
```

The result is as below:

start run exectue
[20:14:08]  COMMAND     POST     "/wd/hub/session/3c5a063f-5181-46fb-acae-9c6d01dd8639/execute"
[20:14:08]  DATA                {"script":"return (function (a, b, c, d) {\r\n                return a + b + c + d;\r\n            }).apply(null, arguments)","args":[1,2,3,4]}
[20:14:08]  COMMAND     GET      "/wd/hub/session/3c5a063f-5181-46fb-acae-9c6d01dd8639/screenshot"
[20:14:08]  DATA                {}
[20:14:08]  RESULT              "iVBORw0KGgoAAAANSUhEUgAABfAAAALMCAYAAACrJZMvAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAP+lSURBVHhe7P15uG1XVeeNn39y03eQDqOhSYKEEIMhoRWB2Jb0Tam8j0UT9OevhKBVPoCgGONbVkFAnqJ5Sjq ... (167498 more bytes)
[20:14:08]      Saved screenshot: ERROR_internetexplorer.11.windows_2018-08-03T12-14-08.651Z.png
{ Error: Remote end send an unknown status code.
    at execute(<Function>, 1, 2, 3, 4) -     at C:\Users\dell\AppData\Roaming\npm\node_modules\wdio-mocha-framework\node_modules\wdio-sync\build\index.js:312:3
  details: undefined,
  message: '',
  type: 'RuntimeError',
  seleniumStack:
   { type: 'Unknown',
     message: 'Remote end send an unknown status code.',
     orgStatusMessage: '' },
  shotTaken: true }
resultis:undefined

Could you help me check it?

**Comment by christian-bromann** [comment]
> Could you help me check it?

Please check the IEDriver logs

**Comment by tyler-tanliang** [comment]
D:\YGLCode\driver>java -Dwebdriver.internetexplorer.driver="D:\YGLCode\driver\IEDriverServer.exe"  -jar "D:\YGLCode\driver\selenium-server-standalone-3.13.0.jar"  -role node -hub http://127.0.0.1:4444/grid/register -browser "browserName=internet explorer,version=11,platformName=windows,platform=windows"
20:28:42.645 INFO [GridLauncherV3.launch] - Selenium build info: version: '3.13.0', revision: '2f0d292'
20:28:42.664 INFO [GridLauncherV3$3.launch] - Launching a Selenium Grid node on port 1296
2018-08-06 20:28:42.988:INFO::main: Logging initialized @755ms to org.seleniumhq.jetty9.util.log.StdErrLog
20:28:43.193 INFO [SeleniumServer.boot] - Selenium Server is up and running on port 1296
20:28:43.193 INFO [GridLauncherV3$3.launch] - Selenium Grid node is up and ready to register to the hub
20:28:43.323 INFO [SelfRegisteringRemote$1.run] - Starting auto registration thread. Will try to register every 5000 ms.
20:28:43.323 INFO [SelfRegisteringRemote.registerToHub] - Registering the node to the hub: http://127.0.0.1:4444/grid/register
20:28:44.856 INFO [SelfRegisteringRemote.registerToHub] - The node is registered to the hub and ready to use
20:29:33.891 INFO [ActiveSessionFactory.apply] - Capabilities are: {
  "browserName": "internet explorer",
  "handlesAlerts": true,
  "javascriptEnabled": true,
  "locationContextEnabled": true,
  "loggingPrefs": {
    "browser": "ALL",
    "driver": "ALL"
  },
  "platform": "WINDOWS",
  "requestOrigins": {
    "url": "http:\u002f\u002fwebdriver.io",
    "version": "4.12.0",
    "name": "webdriverio"
  },
  "rotatable": true,
  "seleniumProtocol": "WebDriver",
  "version": ""
}
20:29:33.897 INFO [ActiveSessionFactory.lambda$apply$11] - Matched factory org.openqa.selenium.remote.server.ServicedSession$Factory (provider: org.openqa.selenium.ie.InternetExplorerDriverService)
Started InternetExplorerDriver server (32-bit)
3.13.0.0
Listening on port 27128
Only local connections are allowed
20:29:36.197 INFO [ProtocolHandshake.createSession] - Detected dialect: W3C
20:29:36.245 INFO [RemoteSession$Factory.lambda$performHandshake$0] - Started new session 14898747-9a90-4b55-a18d-d4063ad346d3 (org.openqa.selenium.ie.InternetExplorerDriverService)
20:29:38.682 INFO [ActiveSessions$1.onStop] - Removing session 14898747-9a90-4b55-a18d-d4063ad346d3 (org.openqa.selenium.ie.InternetExplorerDriverService)


Above is the IEDriver logs.

**Comment by christian-bromann** [comment]
The logs don't really show anything that could hint why it is not working. I know for a fact this is issue is unrelated to WebdriverIO. Something is up in the IEDriver or the setup. Can't really tell what it is. Please reach out to the IEDriver folks. Thanks!

**Comment by tyler-tanliang** [comment]
It's strange I can run execute with C# and IEDriver in selenium grid.
I will try in another computer. Maybe it's an environment issue.
Thanks very much!

**Comment by christian-bromann** [comment]
Compare the logs .. maybe they differ

**Comment by tyler-tanliang** [comment]
Thanks, I will prepare a new one computer today and then try again.

**Comment by tyler-tanliang** [comment]
When I used the iedriver and selenium with the version 3.14 on different computer, there is another message about the issue.
{ Error: An unknown server-side error occurred while processing the command.
    at execute(<Function>, 1, 2, 3, 4) -     at C:\Users\dell\AppData\Roaming\npm\node_modules\wdio-mocha-framework\node_modules\wdio-sync\build\index.js:312:3
  details: undefined,
  message: 'java.io.IOException: Server returned HTTP response code: 405 for URL: http://localhost:47300/session/2bb6336a-f1e9-45dd-8e83-4a52903ea73d/execute',
  type: 'RuntimeError',
  seleniumStack:
   { type: 'UnknownError',
     message: 'An unknown server-side error occurred while processing the command.',
     orgStatusMessage: 'java.io.IOException: Server returned HTTP response code: 405 for URL: http://localhost:47300/session/2bb6336a-f1e9-45dd-8e83-4a52903ea73d/execute' },
  shotTaken: true }

**Comment by savsav** [comment]
@tyler-tanliang , https://github.com/webdriverio/webdriverio/pull/2846 has fixed the issue with execute and other commands. However, there are other outstanding issues with IE11 - either with the IE driver or with WebdriverIO.

I am currently trying to make it work for a site but it keeps failing in different ways. Some advices:

- don't set cookies using the selenium api, go for javascript instead (will be fixed in iedriver 3.15)
- don't use arrow functions as IE11 doesn't support them
- make sure your selectors match only one element on the page - until #2840 is sorted out

**Comment by tyler-tanliang** [comment]
When I used the latest version about WDIO 4.13.2 with IEDriver version 3.13.
It can work well now.
